tools/kleaver/main.cpp

```diff
@@ -95,6 +95,11 @@ namespace {
   llvm::cl::opt<std::string> directoryToWriteQueryLogs("query-log-dir",llvm::cl::desc("The folder to write query logs to. Defaults is current working directory."),
 		                                               llvm::cl::init("."));
 
+  llvm::cl::opt<bool>
+  QueryIndependence("assume-query-independence",
```

Comment:
As an additional note, I think it's better to have the option name refer to the high-level assumption (that the queries are independent to one another), rather than to new terms such as "array symbol table", which users should not be concerned with.